admission/hooks/validate_httpproxy.go

```diff
@@ -30,11 +30,12 @@ func (v *contourHTTPProxyValidator) Handle(ctx context.Context, req admission.Re
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 	newAnn := hp.GetAnnotations()
+	newIngressClassNameField, _, _ := getHTTPProxyIngressClassNameField(hp)
```

Comment:
Check err and ok.

Reply:
In case of annotations, nonexistent and empty value are not discriminated (= `ok` is not used). I will do so in case of `ingressClassName` field too.

Reply:
It's OK to equate `newIngressClassNameField == ""` to `!ok`.
What I mean is that there is no guarantee `newIngressClassNameField` is equal to `""` when `ok` is false.

In case of maps, the Go language guarantees that.
https://golang.org/doc/effective_go#maps
> An attempt to fetch a map value with a key that is not present in the map will return the zero value for the type of the entries in the map.